lib/puppet/util/windows/adsi.rb

```diff
@@ -302,7 +365,7 @@ def initialize(name, native_group = nil)
     end
 
     def uri
-      self.class.uri(name)
+      self.class.uri(*self.class.parse_name(name))
```

Comment:
Brings the same behavior for user domain\account checking that was present in users.